index.js

```diff
@@ -118,93 +149,146 @@ async function isNotLatestTerraformVersion() {
+        matchingProject.headSha
+      );
+      matchingProject.oldSha = getSha(image.newName);
+      matchingProject.oldUrl = image.newName;
+      image.newName = matchingProject.headUrl;
+      return matchingProject;
+    })
+  );
+}
 
-  if (newBlob !== data.content) {
-    closePRs();
+// Main ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-    const adminMsgs = await getCommitMessages(
-      "notification-admin",
-      oldAdminSha // eslint-disable-line no-undef
-    );
-    const apiMsgs = await getCommitMessages("notification-api", oldApiSha); // eslint-disable-line no-undef
+async function run() {
+  const releaseContent = await getContents(
+    "cds-snc",
+    "notification-manifests",
+    "env/production/kustomization.yaml"
+  );
 
-    let logs = `ADMIN: \n\n ${adminMsgs.join(
-      "\n"
-    )} \n\n API: \n\n ${apiMsgs.join("\n")}`;
-    if (await isNotLatestManifestsVersion()) {
-      logs = `⚠️ **The production version of manifests is behind the latest staging version. Consider upgrading to the latest version before merging this pull request.** \n\n ${logs}`;
-    }
+  const prTemplate = await getContents(
+    "cds-snc",
+    "notification-manifests",
+    ".github/PULL_REQUEST_TEMPLATE.md"
+  );
 
-    if (await isNotLatestTerraformVersion()) {
-      logs = `⚠️ **The production version of the Terraform infrastructure is behind the latest staging version. Consider upgrading to the latest version before merging this pull request.** \n\n ${logs}`;
-    }
+  const releaseConfig = YAML.parse(Base64.decode(releaseContent.content));
+  const issueContent = Base64.decode(prTemplate.content);
 
-    const branchName = `release-${new Date().getTime()}`;
-
-    await octokit.git.createRef({
-      owner: "cds-snc",
-      repo: "notification-manifests",
-      ref: `refs/heads/${branchName}`,
-      sha: manifestsSha,
-    });
-
-    await octokit.repos.createOrUpdateFile({
-      owner: "cds-snc",
-      repo: "notification-manifests",
-      branch: branchName,
-      sha: data.sha,
-      path: "env/production/kustomization.yaml",
-      message: `Updated manifests to admin:${adminSha.slice(
-        0,
-        7
-      )} and api:${apiSha.slice(0, 7)}`,
-      content: newBlob,
-    });
-
-    await octokit.pulls.create({
-      owner: "cds-snc",
-      repo: "notification-manifests",
-      title: `[AUTO-PR] Automatically generated new release ${new Date().toISOString()}`,
-      head: branchName,
-      base: "main",
-      body: issueContent.replace(
-        "> Give details ex. Security patching, content update, more API pods etc",
-        logs
-      ),
-      draft: true,
-    });
+  // Build up projects and update images with latest SHAs.
+  await hydrateWithSHAs(releaseConfig, PROJECTS);
+
+  // Return if no new changes.
+  const newReleaseContentBlob = Base64.encode(YAML.stringify(releaseConfig));
+  if (newReleaseContentBlob.trim() === releaseContent.content.trim()) {
```

Comment:
There was a minor bug here. Without the `trim()` call, it would never be equal because the version fetched from the repository had a new line and the one built on the fly with updated image SHAs never did.

Reply:
Ah, thank you for fixing this! Noticed sometimes that PRs were opened without changes and this was not ideal

Reply:
I haven't tested with a scenario where that would apply so if it still happen, I can take a look. But this was definitely an issue that would prevent this from working as intended.